slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/AttestationWatermarkUpdater.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection.interchange;
+
+import tech.pegasys.web3signer.slashingprotection.dao.LowWatermarkDao;
+import tech.pegasys.web3signer.slashingprotection.dao.SigningWatermark;
+import tech.pegasys.web3signer.slashingprotection.dao.Validator;
+
+import java.util.Optional;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.units.bigints.UInt64;
+import org.jdbi.v3.core.Handle;
+
+public class AttestationWatermarkUpdater {
```

Comment:
Just a though.. Thinking we probably don't need this class. The source and target are now independent watermarks values so don't need to be grouped together. And there is common logic between all the watermarks such as the findBest checks.

Reply:
so all of this has been pulled into the importer - and I THINK its too big ... but see what you think